src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,70 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import debounce from 'lodash.debounce';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as filterActions } from '../../features/filter';
+import { FilterStatusType } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const filterParams = useAppSelector(state => state.filter);
+
+  const dispatch = useAppDispatch();
+
+  const setFilterQuery = (newQuery: string) => {
+    dispatch(filterActions.setQuery(newQuery));
+  };
+
+  const debouncedSetFilterQuery = useCallback(
+    debounce(setFilterQuery, 300),
+    [],
+  );
+
+  const handleQueryInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.currentTarget.value);
+    debouncedSetFilterQuery(event.currentTarget.value);
+  };
+
+  const handleQueryClear = () => {
+    setQuery('');
+    debouncedSetFilterQuery('');
```

Comment:
I think you should reset filtered list immediately after clicking on `clearSearchButton`, not after delay 300ms

Reply:
Yeap, it would be great